### PR TITLE
chore(gatsby-cli): Convert util/configstore to typescript

### PR DESCRIPTION
**Pull request by mottox2**
## Description

Convert `util/configstore` to typescript

## Related Issues

Related to #21995

### CODE REVIEW

File: packages/gatsby-cli/src/util/configstore.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  // This should never happen (?)
+  conf = {
+    settings: {
+      "cli.packageManager": ``,
+    },
+    get: (key: string): string => conf.settings![key],
+    set: (key: string, value: string): void => {
+      conf.settings![key] = value
+    },
+  }
```
Comment (mottox2) [inline packages/gatsby-cli/src/util/configstore.ts]: It seems that this should never happen.`Configstore` is also used in gatsby-core-utils, but try-catch is not used. I would like to delete these lines.

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-core-utils/src/get-config-store.ts

Comment (blainekasten) [reply packages/gatsby-cli/src/util/configstore.ts]: Yeah, looked at configstore's code. It doesn't ever throw on instantiation. Let's use the `gatsby-core-utils` version and drop the try/catch stuff

File: packages/gatsby-cli/src/util/configstore.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+const packageMangerConfigKey = `cli.packageManager`
+export const getPackageManager = (): string => conf.get(packageMangerConfigKey)
+export const setPackageManager = (packageManager: string): void => {
+  conf.set(packageMangerConfigKey, packageManager)
+  report.info(`Preferred package manager set to "${packageManager}"`)
+}
```
Comment (blainekasten) [inline packages/gatsby-cli/src/util/configstore.ts]: these string types can be tightened to just being `'yarn' | 'npm'`

**Comment by mottox2** [comment]
@blainekasten 
Thanks! I use `gatsby-core-utils/get-config-store`.
Should I rename this file to `util/package-manager.ts`?

**Comment by blainekasten** [comment]
@mottox2 I like that idea of renaming it to `package-manager` if you could update all the references too. I'll wait for that then merge

**Pull request merged**